database/memdb/db.go

```diff
@@ -38,6 +39,21 @@ func New() *Database {
 	return NewWithSize(DefaultSize)
 }
 
+// Copy returns a Database with the same key-value pairs as db
+func Copy(db *Database) (*Database, error) {
```

Comment:
We need this utility to be able to copy the memdb between VM instances, since closing a database makes it so that it's unusable afterwards.